### PR TITLE
deps: update to react-spring@9

### PR DESCRIPTION
**Pull request by williaster**
Duplicates #1218 with fixes for merge conflicts + bumping to `react-spring@^9.2.0` to see if types are fixed and potentially leverage in #1268 

cc @gazcn007 @garfieldduck

### CODE REVIEW

File: packages/visx-xychart/package.json [hunk 1]
```diff
@@ -37,7 +37,7 @@
   },
   "peerDependencies": {
     "react": "^16.4.0-0",
-    "react-spring": "^8.0.27"
+    "react-spring": "^9.0.0-rc.3"
```
Comment (gazcn007) [inline packages/visx-xychart/package.json:40]: this should be 9.2.0 as well right?

Comment (williaster) [reply packages/visx-xychart/package.json]: yep, thanks!

**Comment by williaster** [comment]
Noting this closes #1122

**Review by kristw — APPROVED** [review]

**Comment by williaster** [comment]
all right, going to merge this which means we'll be in alpha for the next few releases 🤞

**Pull request merged**